### PR TITLE
KNOX-2228 - JWTFilter should handle unknown token exception from toke…

### PR DESCRIPTION
**Pull request by pzampino**
…n state service

## What changes were proposed in this pull request?

Modified handling of "Unknown token" exception in JWTFilters, such that it no longer results in a HTTP 500 response. Also, improved some token-related log messages.

## How was this patch tested?

Augmented CommonJWTFilterTest.

### CODE REVIEW

File: gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/AbstractJWTFilter.java [hunk 1]
```diff
@@ -170,7 +170,13 @@ protected void configureExpectedParameters(FilterConfig filterConfig) {
   protected boolean tokenIsStillValid(JWT jwtToken) {
     Date expires;
     if (tokenStateService != null) {
-      expires = new Date(tokenStateService.getTokenExpiration(jwtToken.toString()));
+      long timestamp = 0;
+      try {
+        timestamp = tokenStateService.getTokenExpiration(jwtToken.toString());
+      } catch (Exception e) {
```
Comment (risdenk) [inline gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/AbstractJWTFilter.java]: Do we not have a specific exception coming from getTokenExpiration? or tokenStateService?

File: gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/JWTMessages.java [hunk 2]
```diff
@@ -41,6 +41,9 @@
   @Message( level = MessageLevel.INFO, text = "Unable to verify token: {0}" )
   void unableToVerifyToken(@StackTrace( level = MessageLevel.ERROR) Exception e);
 
+  @Message( level = MessageLevel.ERROR, text = "Unable to verify token expiration: {0}" )
```
Comment (risdenk) [inline gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/JWTMessages.java]: Do we need this at an error level?

File: gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/AbstractJWTFilter.java (hunk 1 above)
Comment (pzampino) [reply gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/AbstractJWTFilter.java]: I've been debating the need for an UnknownTokenException, and your comment may have just convinced me to implement it.

File: gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/JWTMessages.java (hunk 2 above)
Comment (pzampino) [reply gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/JWTMessages.java]: I went back and forth with this, and could go either way really. Do you think INFO is sufficient?

Comment (risdenk) [reply gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/JWTMessages.java]: INFO or WARN should be good I think. I just saw the message above about unable to verify being INFO and was like hmmm that seems worse than not being able to verify the expiration.

File: gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/AbstractJWTFilter.java (hunk 1 above)
Comment (pzampino) [reply gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/AbstractJWTFilter.java]: I would rather change the method signatures in a separate commit, so I've created https://issues.apache.org/jira/browse/KNOX-2230 to address this concern.

**Pull request merged**